aptos-move/move-examples/dao/nft_dao/sources/nft_dao.move

```diff
@@ -0,0 +1,1182 @@
+
+    /// DAO already offered for the new admin
+    const EADMIN_ALREADY_OFFERED: u64 = 15;
+
+    /// DAO offer doesn't exist
+    const EADMIN_OFFER_NOT_EXIST: u64 = 16;
+
+    /// Constants
+    const PROPOSAL_PENDING: u8 = 0;
+    const PROPOSAL_RESOLVED_PASSED: u8 = 1;
+    const PROPOSAL_RESOLVED_NOT_PASSED: u8 = 2;
+    const PROPOSAL_RESOLVED_BY_ADMIN: u8 = 3;
+    const PROPOSAL_VETOED_BY_ADMIN: u8 = 4;
+
+    struct DAO has key {
+        /// Name of the DAO
+        name: String,
+        /// The minimum number of total votes (both yes and no) a proposal must have in order to be considered valid. A proposal would still need more yes than no to pass.
+        resolve_threshold: u64,
+        /// The NFT Collection that is used to govern the DAO
+        governance_token: GovernanceToken,
+        /// The voting duration in secs
+        voting_duration: u64,
+        /// Minimum weight for proposal
+        min_required_proposer_voting_power: u64,
+        /// Proposal counter
+        next_proposal_id: u64,
+        /// DAO resource account signer capability
+        dao_signer_capability: SignerCapability,
+        /// DAO admin account
+        admin: address,
+        /// The pending claims waiting for new admin to claim
+        pending_admin: Option<address>,
+    }
+
+    /// The collection should only contain globally unique NFTs
+    /// The total supply is fixed with the token names.
+    struct GovernanceToken has drop, store {
+        /// The creator address of this NFT collection
+        creator: address,
+        /// The collection name
+        collection: String,
+    }
+
+    /// All proposals
+    struct Proposals has key {
+        proposals: Table<u64, Proposal>,
+    }
+
+    /// Store the general information about a proposal
+    struct Proposal has copy, drop, store {
+        /// Name of the proposal, limiting to 64 chars
+        name: String,
+        /// Description of the proposal, limiting to 512 chars
+        description: String,
+        /// The name of function to be executed
+        function_name: String,
+        /// The function arguments to be executed
+        function_args: PropertyMap,
+        /// The start time of the voting
+        start_time_sec: u64,
+        /// Proposal results, unresolved, passed, not passed
+        resolution: u8,
+        /// final voting count of yes votes
+        final_yes_votes: u64,
+        /// final voting count of no votes
+        final_no_votes: u64,
+    }
+
+    struct ProposalVotingStatistics has key {
+        proposals: Table<u64, VotingStatistics>,
+    }
+
+    struct VotingStatistics has store {
+        /// Total yes votes
+        total_yes: u64,
+        /// Total no notes
+        total_no: u64,
+        /// Token voted yes
+        yes_votes: BucketTable<TokenId, address>, // address is the original voter's address for keeping a record of who voted
+        /// Token voted no
+        no_votes: BucketTable<TokenId, address>,
```

Comment:
It isn't clear why we need to track the address up until this point. Perhaps we could add some notes about the design and constraints at the top.